src/adapter/receiver.rs

```diff
@@ -68,25 +69,36 @@ fn extract_kind_string(ty: &Type) -> String {
                                         } => {
                                             let inner = extract_kind_string(type_);
                                             if *is_mutable {
-                                                format!("&mut {}", inner)
+                                                Cow::Owned(format!("&mut {}", inner))
                                             } else {
-                                                format!("&{}", inner)
+                                                Cow::Owned(format!("&{}", inner))
                                             }
                                         }
                                         _ => extract_kind_string(t),
                                     }
                                 }
-                                rustdoc_types::GenericArg::Lifetime(lt) => lt.clone(),
-                                rustdoc_types::GenericArg::Const(c) => c.expr.clone(),
-                                _ => "?".to_string(),
+                                rustdoc_types::GenericArg::Lifetime(lt) => {
+                                    Cow::Borrowed(lt.as_str())
+                                }
+                                rustdoc_types::GenericArg::Const(c) => {
+                                    Cow::Borrowed(c.expr.as_str())
+                                }
+                                _ => unreachable!("should not encounter infer"),
                             })
                             .collect();
-                        format!("{}<{}>", name, args_str.join(", "))
+
+                        let args_joined = args_str
+                            .iter()
+                            .map(|cow| cow.as_ref())
+                            .collect::<Vec<_>>()
```

Comment:
Do we have to `.collect()` first to join, or can we join directly without allocating first?

Reply:
yes we need to `.collect()`